app/jobs/parse_report_job.rb

```diff
@@ -32,7 +32,8 @@ def save_all
     notify_payload_tracker(:processing, "Job #{jid} is now processing")
     parser.save_all
     notify_payload_tracker(:success, "Job #{jid} has completed successfully")
-  rescue ::MissingIdError, ::WrongFormatError, ::InventoryHostNotFound => e
+  rescue ::MissingIdError, ::WrongFormatError, ::InventoryHostNotFound,
+         ::ActiveRecord::RecordInvalid => e
```

Comment:
We don't want to push the job back to redis if the record is invalid (i.e. someone uploads the same report twice).

Reply:
Can't we be more deterministic and assert before parsing if we already are parsing the incoming report?

Reply:
We could check for specific failures, like if a TestResult already exists, before trying to save; however, if a validation error is raised in any of the model transactions, we don't want to continue to retry, because the same report upload will just fail again (and we'll try it 3 times before removing the job from redis). By rescuing this exception, we will never retry this kind of errored report upload.